packages/connection-form/src/components/connection-form.tsx

```diff
@@ -97,6 +108,165 @@ type ConnectionFormPropsWithoutPreferences = {
   onSaveConnectionClicked: (connectionInfo: ConnectionInfo) => Promise<void>;
 };
 
+const colorPreviewStyles = css({
+  height: '16px',
+  width: '16px',
+  marginRight: spacing[2],
+});
+
+const ColorCircleGlyph = createGlyphComponent('ColorCircle', (props) => (
+  <svg
+    {...props}
+    className={colorPreviewStyles}
+    width="16"
+    height="16"
+    viewBox="0 0 16 16"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <g clip-path="url(#clip0_756_18092)">
+      <line
+        opacity="0.5"
+        x1="16.3536"
+        y1="-0.146447"
+        x2="0.353554"
+        y2="15.8536"
+        stroke="#DB3030"
+      />
+    </g>
+    <rect
+      x="0.5"
+      y="0.5"
+      width="15"
+      height="15"
+      rx="7.5"
+      stroke="#889397"
+      fill={props.hexColor}
+    />
+    <defs>
+      <clipPath id="clip0_756_18092">
+        <rect width="16" height="16" rx="8" fill="white" />
+      </clipPath>
+    </defs>
+  </svg>
+));
+
+const personalisationSectionLayoutStyles = css({
+  display: 'grid',
+  gridTemplateColumns: '1fr 1fr',
+  gridTemplateRows: 'auto',
+  gridTemplateAreas: `
+    'name-input color-input'
+    'favorite-marker favorite-marker'
+  `,
+  gap: spacing[4],
+  marginBottom: spacing[4],
+});
+
+type ConnectionPersonalisationFormProps = {
+  initialValue?: ConnectionPersonalisationOptions;
+  onChange: (ConnectionPersonalisationOptions) => void;
+};
+
+function ConnectionPersonalisationForm({
+  onChange,
+  initialValue,
+}: ConnectionPersonalisationFormProps): React.ReactElement {
+  const showFavoriteActions = useConnectionFormPreference(
+    'showFavoriteActions'
+  );
+
+  const [name, setName] = useState(initialValue?.name || '');
+  const [isNameDirty, setNameDirty] = useState(!!initialValue?.isNameDirty);
+  const [color, setColor] = useState(initialValue?.color || undefined);
+  const [isFavorite, setFavorite] = useState(initialValue?.isFavorite);
```

Comment:
You are pushing all this state to the connection form state higher in the tree, why do we need to keep a copy here in the local state?

Reply:
Yeah, this is something I was drafting before moving everything to the reducer. The new version uses only the reducer state and I got rid of these useState.